fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/StorageView.java

```diff
@@ -62,4 +62,16 @@
 	 * or an estimate of the number of resources that could be stored if this view has a blank resource.
 	 */
 	long getCapacity();
+
+	/**
+	 * If this is view is a delegate around another storage view, return the underlying view.
+	 * This can be used to check if two views refer to the same inventory "slot".
+	 * <b>Do not try to extract from the underlying view, or you risk bypassing some checks.</b>
+	 *
+	 * <p>It is expected that two storage views with the same underlying view (a.getUnderlyingView() == b.getUnderlyingView())
+	 * share the same content, and mutating one should mutate the other. However, one of them may allow extraction, and the other may not.
+	 */
+	default StorageView<T> getUnderlyingView() {
```

Comment:
Does this have any other use besides for deduplicating views? I think it's better to use a marker type and return that instead (sealed interface maybe? Is that a thing?)

Reply:
No, it's the only use. I agree that there could be misuse, but this is quite niche, and I think that a marker type would only make the API even more confusing than it already is?

Reply:
Hmm, fair enough :+1: